### ISSUE TITLE
Make route keywords configureable

### ISSUE THREAD

**Issue by kirrg001** [TryGhost/Ghost#8492]
I am raising this as a community issue. I think it would be helpful for some people to make route keywords configureable. 

See https://github.com/TryGhost/Ghost/blob/master/core/server/config/overrides.json#L21

Right now you are not allowed to override the route keywords (e.g. `/subscribe`). If you move the route keywords into the `defaults.json`, it would be allowed to add e.g. 

```
"routeKeywords": {
  "subscribe": "something"
}
``` 

into your custom configuration file e.g. `config.production.json`. And the result would be `/something` instead of `/subscribe`.

I've already tested it for the subscriber route, worked pretty good.  But this needs more testing for all route key words and probably more tests.

Ghost: master only

**Comment by ErisDS** [comment]
The reason this hasn't been done is due to the question of how: should it be config, or a setting, or part of whatever mechanism we provide for configuring channels, or part of i18n, or something else?

This was placed in config as the quickest way to centralise it and stop having the same strings all over the codebase, but not much more thought went into how we want to expose this to users/developers.

Originally, we were stuck in a conflict with this an many other things, because we do not want to have lots of settings in the UI, but config is not user-friendly. More recently we've discussed exposing features in layers, or with escape valves, so that as features become more complex they get harder to access, and require a developer instead of a user:

- sensible defaults (users)
- settings (users)
- exposed config/code e.g. code injection (users)
- config (developers)
- hacking core (developers)

There's also apps and themes in the mix 😬

TLDR; I don't think this is a question of "should we do it" but "_how_ should we do it". There hasn't really been demand and because this config interacts very tightly with both channels and internal apps, I am concerned that opening it up without clarity about how could get us backed into an awkward corner.

Just my always-cautious 2c 🙃

**Comment by ErisDS** [comment]
Closing this now in favour of #9192, this should fall out as a result of the refactoring of routes.
